### PR TITLE
chore: file movement for two fa attempts inside dashboard

### PR DESCRIPTION
**Pull request by Riddhiagrawal001**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Chore changes to introduce totp and recovery code attempts expired inside the dashboard .

## How did you test it?
Change totp and regenerate recovery codes should work same 

## Where to test it?
- [X] INTEG
- [X] SANDBOX
- [ ] PROD

## Checklist

- [ ] I ran `npm run re:build`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/Settings/HSwitchProfile/TwoFaSettings/TwoFaHelper.res [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                setRecoveryCode(_ => "")
+                setErrorMessage(_ => "")
+                setTwoFaState(_ => Totp)
+              }}>
+              {"Use totp instead"->React.string}
+            </span>
+          </p>
+        </>
+      }}
+      <RenderIf condition={errorMessage->String.length > 0}>
```
Comment (PritishBudhiraja) [inline src/screens/Settings/HSwitchProfile/TwoFaSettings/TwoFaHelper.res:55]: empty string func can be used

File: src/screens/Settings/HSwitchProfile/TwoFaSettings/TwoFaHelper.res [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+          <p className={`${p2Regular} text-jp-gray-700`}>
+            {"Didn't get a code? "->React.string}
+            <span
+              className="cursor-pointer underline underline-offset-2 text-blue-600"
+              onClick={_ => {
+                setRecoveryCode(_ => "")
+                setErrorMessage(_ => "")
+                setTwoFaState(_ => Totp)
+              }}>
+              {"Use totp instead"->React.string}
```
Comment (PritishBudhiraja) [inline src/screens/Settings/HSwitchProfile/TwoFaSettings/TwoFaHelper.res:50]: can be written outside for better read. i feel

File: src/screens/Settings/HSwitchProfile/TwoFaSettings/TwoFaHelper.res [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+          <TwoFaElements.TotpInput otp setOtp />
+          <RenderIf condition={!showOnlyTotp}>
+            <p className={`${p2Regular} text-jp-gray-700`}>
+              {"Didn't get a code? "->React.string}
+              <span
+                className="cursor-pointer underline underline-offset-2 text-blue-600"
+                onClick={_ => {
+                  setOtp(_ => "")
+                  setErrorMessage(_ => "")
+                  setTwoFaState(_ => RecoveryCode)
```
Comment (PritishBudhiraja) [inline src/screens/Settings/HSwitchProfile/TwoFaSettings/TwoFaHelper.res:30]: same make this func outside and just call here

File: src/screens/Settings/HSwitchProfile/TwoFaSettings/ResetTotp.res [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+    } else {
+      Window.removeEventListener("keyup", handleKeyUp)
+    }
+
+    Some(
+      () => {
+        Window.removeEventListener("keyup", handleKeyUp)
+      },
+    )
+  }, [otpInModal, recoveryCode])
```
Comment (PritishBudhiraja) [inline src/screens/Settings/HSwitchProfile/TwoFaSettings/ResetTotp.res:146]:
```suggestion
  React.useEffect(() => {
    otpInModal->String.length == 6 || recoveryCode->String.length == 9
      ? Window.addEventListener("keyup", handleKeyUp)
      : Window.removeEventListener("keyup", handleKeyUp)

    Some(_ => Window.removeEventListener("keyup", handleKeyUp))
  }, [otpInModal, recoveryCode])
```

File: src/screens/Settings/HSwitchProfile/TwoFaSettings/ResetTotp.res [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+    }
+    None
+  }, [])
+
+  let handleKeyUp = ev => {
+    open ReactEvent.Keyboard
+    let key = ev->key
+    let keyCode = ev->keyCode
+
+    if key === "Enter" || keyCode === 13 {
```
Comment (PritishBudhiraja) [inline src/screens/Settings/HSwitchProfile/TwoFaSettings/ResetTotp.res:129]: only keyCode is sufficent to check I feel as keyCode is unique

File: src/screens/Settings/HSwitchProfile/TwoFaSettings/ResetTotp.res [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+      handle2FaVerify()->ignore
+    }
+  }
```
Comment (PritishBudhiraja) [inline src/screens/Settings/HSwitchProfile/TwoFaSettings/ResetTotp.res:132]: only keyCode is sufficent to check I feel as keyCode is unique
```suggestion
  let handleKeyUp = ev => {
    let keyCode = ev->ReactEvent.Keyboard.keyCode
    if keyCode === 13 {
      handle2FaVerify()->ignore
    }
  }
```

File: src/screens/Settings/HSwitchProfile/TwoFaSettings/ResetTotp.res (hunk 4 above)
Comment (Riddhiagrawal001) [reply src/screens/Settings/HSwitchProfile/TwoFaSettings/ResetTotp.res:146]: Ternary looks non readable hence made it if and else

File: src/screens/Settings/HSwitchProfile/TwoFaSettings/ResetTotp.res (hunk 6 above)
Comment (Riddhiagrawal001) [reply src/screens/Settings/HSwitchProfile/TwoFaSettings/ResetTotp.res:132]: By using both checks (key === 'Enter' || keyCode === 13), we ensure that your application works across all browsers, and for older and newer versions

File: src/screens/Settings/HSwitchProfile/TwoFaSettings/RegenerateRC.res [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+    } else {
+      Window.removeEventListener("keyup", handleKeyUp)
+    }
+
+    Some(
+      () => {
+        Window.removeEventListener("keyup", handleKeyUp)
+      },
+    )
+  }, [otpInModal])
```
Comment (PritishBudhiraja) [inline src/screens/Settings/HSwitchProfile/TwoFaSettings/RegenerateRC.res:98]:
```suggestion
  React.useEffect(() => {
    otpInModal->String.length == 6
      ? Window.addEventListener("keyup", handleKeyUp)
      : Window.removeEventListener("keyup", handleKeyUp)

    Some(_ => Window.removeEventListener("keyup", handleKeyUp))
  }, [otpInModal])
```

File: src/screens/Settings/HSwitchProfile/TwoFaSettings/RegenerateRC.res [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+
+  let handleKeyUp = ev => {
+    open ReactEvent.Keyboard
+    let key = ev->key
+    let keyCode = ev->keyCode
+
+    if key === "Enter" || keyCode === 13 {
+      verifyTOTP()->ignore
+    }
+  }
```
Comment (PritishBudhiraja) [inline src/screens/Settings/HSwitchProfile/TwoFaSettings/RegenerateRC.res:84]: Same just check once

**Review by PritishBudhiraja — APPROVED** [review]

File: src/screens/Settings/HSwitchProfile/TwoFaSettings/RegenerateRC.res (hunk 7 above)
Comment (Riddhiagrawal001) [reply src/screens/Settings/HSwitchProfile/TwoFaSettings/RegenerateRC.res:98]: Ternary looks non readable hence made it if and else

File: src/screens/Settings/HSwitchProfile/TwoFaSettings/RegenerateRC.res (hunk 8 above)
Comment (Riddhiagrawal001) [reply src/screens/Settings/HSwitchProfile/TwoFaSettings/RegenerateRC.res:84]: By using both checks (key === 'Enter' || keyCode === 13), we ensure that your application works across all browsers, and for older and newer versions

**Review by JeevaRamu0104 — APPROVED** [review]

**Pull request merged**